### PR TITLE
feat: add webrtc-direct listener in go-peer

### PR DESCRIPTION
**Pull request by 2color**
## TODO

- [x] Update go-libp2p deps

### CODE REVIEW

File: go-peer/main.go [hunk 1]
```diff
@@ -169,7 +170,10 @@ func main() {
 		libp2p.Identity(privk),
 		libp2p.Transport(quicTransport.NewTransport),
 		libp2p.Transport(webtransport.New),
+		libp2p.Transport(webrtc.New),
 		libp2p.ListenAddrStrings("/ip4/0.0.0.0/udp/9095/quic-v1", "/ip4/0.0.0.0/udp/9095/quic-v1/webtransport"),
+		// 👇 webrtc-direct cannot listen on the same port as QUIC or WebTransport
+		libp2p.ListenAddrStrings("/ip4/0.0.0.0/udp/9096/webrtc-direct"),
```
Comment (2color) [inline go-peer/main.go:176]: In the latest version of go-libp2p, the same port can be reused for WebRTC, QUIC, and WebTransport, and even TCP. 
```suggestion
		libp2p.ListenAddrStrings("/ip4/0.0.0.0/udp/9095/webrtc-direct"),
```

Comment (MarcoPolo) [reply go-peer/main.go:176]: And let's remove the comment as well

File: go-peer/main.go [hunk 2]
```diff
@@ -169,7 +170,10 @@ func main() {
 		libp2p.Identity(privk),
 		libp2p.Transport(quicTransport.NewTransport),
 		libp2p.Transport(webtransport.New),
+		libp2p.Transport(webrtc.New),
 		libp2p.ListenAddrStrings("/ip4/0.0.0.0/udp/9095/quic-v1", "/ip4/0.0.0.0/udp/9095/quic-v1/webtransport"),
+		// 👇 webrtc-direct cannot listen on the same port as QUIC or WebTransport
```
Comment (2color) [inline go-peer/main.go:175]:
```suggestion
```

**Comment by 2color** [comment]
Rust tests are failing. A good chance to update the workflow to only run the rust build for changes in the rust code

**Pull request merged**